benchmark/Makefile

```diff
@@ -0,0 +1,16 @@
+all: main
+
+main:
+	clang++ -std=c++14 \
```

Comment:
Dose `-O3` flag contribute to the benchmark? We could have a try.

Reply:
I agree. I think -O3 could be more useful when used to build `libcgotorch.so`.

Reply:
I got no significant improvement on my host, average the throughput of 5 times running.

Enable `-O3`
``` bash
➜  gotorch git:(mnist_benchmark) ✗ make -C benchmark
clang++ -std=c++14 -O3 \
	-I /Users/yancey/go/src/github.com/wangkuiyi/gotorch/cgotorch/libtorch/include \
	-I /Users/yancey/go/src/github.com/wangkuiyi/gotorch/cgotorch/libtorch/include/torch/csrc/api/include \
	-L /Users/yancey/go/src/github.com/wangkuiyi/gotorch/cgotorch/libtorch/lib \
	main.cpp \
	-o mnist \
	-Wl,-rpath,/Users/yancey/go/src/github.com/wangkuiyi/gotorch/cgotorch/libtorch/lib \
	-pthread \
	-lc10 -ltorch -ltorch_cpu \
	-D_GLIBCXX_USE_CXX11_ABI=0
➜  gotorch git:(mnist_benchmark) ✗ for i in `seq 5`; do ./benchmark/mnist; done | awk '{sum+=$3} END {print sum/NR}'
18927.3
```

Without `-O3`

``` bash
➜  gotorch git:(mnist_benchmark) make -C benchmark
clang++ -std=c++14 \
	-I /Users/yancey/go/src/github.com/wangkuiyi/gotorch/cgotorch/libtorch/include \
	-I /Users/yancey/go/src/github.com/wangkuiyi/gotorch/cgotorch/libtorch/include/torch/csrc/api/include \
	-L /Users/yancey/go/src/github.com/wangkuiyi/gotorch/cgotorch/libtorch/lib \
	main.cpp \
	-o mnist \
	-Wl,-rpath,/Users/yancey/go/src/github.com/wangkuiyi/gotorch/cgotorch/libtorch/lib \
	-pthread \
	-lc10 -ltorch -ltorch_cpu \
	-D_GLIBCXX_USE_CXX11_ABI=0
➜  gotorch git:(mnist_benchmark) for i in `seq 5`; do ./benchmark/mnist; done | awk '{sum+=$3} END {print sum/NR}'
19361.8
```